src/org/gk/slicing/RevisionDetector.java

```diff
@@ -0,0 +1,442 @@
+			if (revised) {
+				// TODO determine API database UpdateTrack schema class.
+				SlicingEngine.logAndPrintln("Revision detected in: " + inst, ps);
+			}
+		}
+	}
+
+	/**
+	 * Check if Pathway is revised.
+	 *
+	 * TODO Is the "in-method" recursive approach that
+	 * {@link InstanceUtilities#getContainedInstances(GKInstance, String...)} takes recommended?
+	 *
+	 * @param pathway
+	 * @return boolean (true if revised, false otherwise).
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 */
+	private boolean isPathwayRevised(GKInstance pathway, MySQLAdaptor targetDBA)
+			throws InvalidAttributeException, Exception {
+		if (pathway == null)
+			return false;
+
+		// Check if an immediate child Pathway is added or removed
+		if (isChildPathwayAddedOrRemoved(pathway, targetDBA))
+			return true;
+
+		// Finally check for changes in summation text.
+		if (isSummationRevised(pathway, targetDBA))
+			return true;
+
+		// Recursively iterate over events in pathway.
+		Collection<GKInstance> events = pathway.getAttributeValuesList(ReactomeJavaConstants.hasEvent);
+
+		if (events != null && events.size() > 0) {
+			for (GKInstance event : events) {
+				// Pathway
+				if (event.getSchemClass().isa(ReactomeJavaConstants.Pathway))
+					isPathwayRevised(event, targetDBA);
+
+				// RLE
+				else if (event.getSchemClass().isa(ReactomeJavaConstants.ReactionlikeEvent)
+					 && isRLERevised(event, targetDBA))
+					return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Check if a child pathway is added or removed for a given parent.
+	 *
+	 * @param parent
+	 * @return boolean (true if a child pathway is added or removed, false otherwise).
+	 * @throws InvalidAttributeException
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	private boolean isChildPathwayAddedOrRemoved(GKInstance parent, MySQLAdaptor targetDBA)
+			throws InvalidAttributeException, SQLException, Exception {
+
+		Collection<GKInstance> childPathways = getChildPathways(parent);
+		Collection<GKInstance> compareChildPathways = getChildPathways(getInstance(targetDBA, parent));
+
+		if (childPathways.isEmpty() && compareChildPathways.isEmpty())
+			return false;
+
+		// If the number of child pathways is different between the two parents,
+		// then an addition or deletion has taken place.
+		if (childPathways.size() != compareChildPathways.size())
+			return true;
+
+		// List of child pathway IDs.
+		Collection<Long> childPathwayDBIDs = childPathways.stream()
+				.map(attrValue -> attrValue.getDBID())
+				.collect(Collectors.toList());
+
+		// Check if there is a child pathway not associated with both parents.
+		Optional<Long> commonDBID = compareChildPathways.stream()
+				// Get database ID.
+				.map(attrValue -> attrValue.getDBID())
+				// Set match criteria (identical database ID's).
+				.filter(comparePathwayDBID -> childPathwayDBIDs.contains(comparePathwayDBID))
+				// Find match (or null if no match).
+				.findAny();
+
+		if (!commonDBID.isPresent())
```

Comment:
This check is not right: If the new slice has two child pathways: 1, 2, and the old slice has these two child pathways: 1, 3, commDBID should have value, 1. Then !commDBID.isPresent() is false (note !). However, these two pathways still have add/remove!

Reply:
Fixed in [5224afe](https://github.com/reactome/CuratorTool/pull/20/commits/5224afe9bbc88b7d82e68e63a4632c6fa9bbe8a5).